src/desktop/apps/auth2/helpers.ts

```diff
@@ -1,17 +1,65 @@
 import Cookies from 'cookies-js'
-import { ModalType } from '@artsy/reaction/dist/Components/Authentication/Types'
+import {
+  ModalType,
+  ModalOptions,
+} from '@artsy/reaction/dist/Components/Authentication/Types'
+import { data as sd } from 'sharify'
+import { pickBy, identity } from 'lodash'
 
 const mediator = require('../../lib/mediator.coffee')
 const LoggedOutUser = require('../../models/logged_out_user.coffee')
 
-export const handleSubmit = (type: ModalType, values, formikBag) => {
+export const handleSubmit = (
+  type: ModalType,
+  modalOptions: ModalOptions,
+  values,
+  formikBag
+) => {
   const user = new LoggedOutUser()
-  user.set(values)
+  const {
+    contextModule,
+    copy,
+    destination,
+    redirectTo,
+    intent,
+    signupReferer,
+    trigger,
+    triggerSeconds,
+  } = modalOptions
```

Comment:
Chatted about this IRL and decided to move the `redirectTo` in this block but that can be addressed in a later PR.